contracts/clearing/BrokerVault.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.27;
+
+import {IVault2} from '../interfaces/IVault2.sol';
+import {TradingApp, ISettle} from './TradingApp.sol';
+import {Ownable2Step} from '@openzeppelin/contracts/access/Ownable2Step.sol';
+import {ReentrancyGuard} from '@openzeppelin/contracts/utils/ReentrancyGuard.sol';
+import {NitroUtils} from '../nitro/libraries/NitroUtils.sol';
+
+contract BrokerVault is IVault2, ISettle, Ownable2Step, ReentrancyGuard {
+	/// @dev Using SafeERC20 to support non fully ERC20-compliant tokens,
+	/// that may not return a boolean value on success.
+	using SafeERC20 for IERC20;
+
+	mapping(address token => uint256 balance) internal _balances;
+	TradingApp internal tradingApp;
+	mapping(uint256 channel_id => bool done) internal performedSettlements;
```

Comment:
Actually, I would make them public as there is no need to hide this data.

Reply:
65098a0